src/quote/swap/adapters/uniswap/index.ts

```diff
@@ -0,0 +1,144 @@
+import { getTokenData, getTokenDataByAddress } from '@indexcoop/tokenlists'
+import { Token, TradeType } from '@uniswap/sdk-core'
+import {
+  AlphaRouter,
+  CurrencyAmount,
+  SwapRoute,
+} from '@uniswap/smart-order-router'
+
+import { AddressZero, EthAddress } from 'constants/addresses'
+import {
+  SwapQuote,
+  SwapQuoteProvider,
+  SwapQuoteRequest,
+} from 'quote/swap/interfaces'
+import { getRpcProvider } from 'utils/rpc-provider'
+import { Exchange, isSameAddress } from 'utils'
+
+function changeToWethIfNecessary(token: string, chainId: number): string {
+  if (isSameAddress(token, EthAddress)) {
+    const weth = getTokenData('WETH', chainId)
+    return weth?.address ?? token
+  }
+  return token
+}
+
+export class UniswapSwapQuoteProvider implements SwapQuoteProvider {
+  constructor(readonly rpcUrl: string) {}
+
+  async getSwapQuote(request: SwapQuoteRequest): Promise<SwapQuote | null> {
+    const { chainId, inputAmount, outputAmount, slippage } = request
+
+    const inputToken = changeToWethIfNecessary(request.inputToken, chainId)
+    const outputToken = changeToWethIfNecessary(request.outputToken, chainId)
+
+    if (isSameAddress(inputToken, outputToken)) {
+      return null
+    }
+
+    if (
+      isSameAddress(inputToken, EthAddress) ||
+      isSameAddress(outputToken, EthAddress)
+    ) {
+      // FIXME: remove for production, just for runnint tests and catching any of these cases
```

Comment:
Flagging for pre-release

Reply:
I'll keep this until we have part 2 because it now actually shouldn't happen anymore but might catch some "bugs".